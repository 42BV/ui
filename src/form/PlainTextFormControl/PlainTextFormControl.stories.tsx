import React from 'react';
import { storiesOf } from '@storybook/react';
import { PlainTextFormControl } from './PlainTextFormControl';
import { Form } from 'reactstrap';

storiesOf('Form/PlainTextFormControl', module).add('basic', () => {
  return (
    <Form>
      <PlainTextFormControl label="Organisation">42 BV</PlainTextFormControl>
    </Form>
  );
});
