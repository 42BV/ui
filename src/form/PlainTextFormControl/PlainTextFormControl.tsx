import { FormGroup, Label } from 'reactstrap';
import React from 'react';
import classNames from 'classnames';

type Props = {
  /**
   * Optionally the label of the form element.
   */
  label?: React.ReactNode;

  /**
   * The value that the form element currently has.
   */
  children: React.ReactNode;

  /**
   * Optional extra CSS class you want to add to the component.
   * Useful for styling the component.
   */
  className?: string;

  /**
   * Optional extra CSS class you want to add to the value.
   * Useful for styling the value.
   */
  valueClassName?: string;

  /**
   * Optional extra CSS class you want to add to the label.
   * Useful for styling the label.
   */
  labelClassName?: string;
};

/**
 * Not all users know that an interaction design element can have
 * a disabled state. Disabled elements have poor contrast and thus
 * are not as readable as they should. Also, screen readers don't
 * give users as much information about the state of the element
 * and keyboard navigation gets a little weird, causing confusion to
 * those using a screen reader. Users might also think the form
 * element might be enabled by changing another form element or in
 * a different use case. So to prevent users confusion or thinking
 * the form is invalid, you should display the value without letting
 * it look like a form element. To prevent alignment issues, the
 * PlainTextFormControl uses spacing as if it were a form element,
 * without the looks of a form element.
 */
export function PlainTextFormControl({
  label,
  children,
  className,
  valueClassName,
  labelClassName
}: Props) {
  const valueClasses = classNames('form-control-plaintext', valueClassName);

  return (
    <FormGroup className={className}>
      {label ? <Label className={labelClassName}>{label}</Label> : null}
      <div className={valueClasses}>{children}</div>
    </FormGroup>
  );
}
