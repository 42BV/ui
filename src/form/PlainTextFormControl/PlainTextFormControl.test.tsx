import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { PlainTextFormControl } from './PlainTextFormControl';

describe('Component: PlainTextFormControl', () => {
  function setup({
    label,
    labelClassName,
    valueClassName
  }: {
    label?: string;
    labelClassName?: string;
    valueClassName?: string;
  }) {
    const props = {
      label,
      labelClassName,
      valueClassName
    };
    const { container } = render(
      <PlainTextFormControl {...props}>This is your name</PlainTextFormControl>
    );
    return { container };
  }

  describe('ui', () => {
    test('default', () => {
      const { container } = setup({});
      expect(container).toMatchSnapshot();
    });

    test('with label', () => {
      setup({
        label: 'Test'
      });
      expect(screen.queryByText('Test')).toBeInTheDocument();
    });

    test('with label class name', () => {
      setup({
        label: 'Test',
        labelClassName: 'label-class-name'
      });
      expect(screen.getByText('Test')).toHaveClass('label-class-name');
    });

    test('with value class name', () => {
      setup({
        valueClassName: 'value-class-name'
      });
      expect(screen.getByText('This is your name')).toHaveClass('value-class-name');
    });
  });
});
