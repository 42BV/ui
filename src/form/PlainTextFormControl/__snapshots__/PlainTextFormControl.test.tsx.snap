// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PlainTextFormControl ui default: Component: PlainTextFormControl => ui => default 1`] = `
<FormGroup
  tag="div"
>
  <Label
    tag="label"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    name
  </Label>
  <div
    className="form-control-plaintext"
  >
    This is your name
  </div>
</FormGroup>
`;

exports[`Component: PlainTextFormControl ui without label: Component: PlainTextFormControl => ui => without label 1`] = `
<FormGroup
  tag="div"
>
  <div
    className="form-control-plaintext"
  >
    This is your name
  </div>
</FormGroup>
`;
